src/globus_sdk/services/tunnels/client.py

```diff
@@ -0,0 +1,330 @@
+from __future__ import annotations
+
+import logging
+import typing as t
+import uuid
+
+from globus_sdk import client, response
+from globus_sdk.scopes import TransferScopes
+
+log = logging.getLogger(__name__)
+
+
+class IterableTunnelResponse(response.IterableResponse):
+    """
+    Response class for non-paged list oriented resources. Allows top level
+    fields to be accessed normally via standard item access, and also
+    provides a convenient way to iterate over the sub-item list in a specified
+    key:
+
+    >>> print("Path:", r["path"])
+    >>> # Equivalent to: for item in r["DATA"]
+    >>> for item in r:
+    >>>     print(item["name"], item["type"])
+    """
+
+    default_iter_key = "data"
+
+
+class TunnelClient(client.BaseClient):
+    r"""
+    Client for the
+    `Globus Transfer API <https://docs.globus.org/api/v2/tunnels/>`_.
+
+    .. sdk-sphinx-copy-params:: BaseClient
+
+    This class provides helper methods for most common resources in the
+    REST API, and basic ``get``, ``put``, ``post``, and ``delete`` methods
+    from the base rest client that can be used to access any REST resource.
+
+    Detailed documentation is available in the official REST API
+    documentation, which is linked to from the method documentation. Methods
+    that allow arbitrary keyword arguments will pass the extra arguments as
+    query parameters.
+    """
+
+    service_name = "transfer"
+    scopes = TransferScopes
+    default_scope_requirements = [TransferScopes.all]
+
+    def create_tunnel(
+        self,
+        listener_stream_ap_id: str,
+        initiator_stream_ap_id: str,
+        submission_id: uuid.UUID | None = None,
```

Comment:
Could this be a `str` containing a UUID?


```suggestion
        submission_id: uuid.UUID | str | None = None,
```